plugins/module_utils/network/sonic/config/lldp_interfaces/lldp_interfaces.py

```diff
@@ -362,16 +450,63 @@ def get_delete_specific_lldp_interfaces_param_requests(self, command, config):
                 if command['tlv_set'].get('ipv6_management_address') is not None:
                     url = self.lldp_intf_config_path['ipv6_management_address'].format(intf_name=name)
                     requests.append({'path': url, 'method': DELETE})
-            if 'tlv_select' in command and command['tlv_select'] is not None:
-                tlv = command['tlv_select'].get('power_management')
-                if tlv:
-                    payload = {"openconfig-lldp-ext:suppress-tlv-advertisement": ["MDI_POWER"]}
-                    url = self.lldp_intf_config_path['suppress_tlv'].format(intf_name=name)
-                    requests.append({'path': url, 'method': PATCH, 'data': payload})
-                else:
-                    url = self.lldp_intf_config_path['suppress_tlv_delete'].format(intf_name=name, med_tlv_select="MDI_POWER")
+            if 'vlan_name_tlv' in command and command['vlan_name_tlv'] is not None:
+                if 'allowed_vlans' in command['vlan_name_tlv'] and command['vlan_name_tlv']['allowed_vlans'] is not None:
+                    allowed_vlan = command['vlan_name_tlv']['allowed_vlans']
```

Comment:
Although SONiC accepts an allowed_vlan argument specified this way, it silently fails on any ranges included in the string. (It accepts them with no error flagged, but doesn't delete them.)

A bug should be submitted against the SONiC code to fix this, but in the meantime, the Ansible implementation would need to break the deletion ranges into comma-separated lists of integer vlan values to make this work correctly.

Please change this logic accordingly. (It would also be worthwhile to submit a Jira bug against SONiC for this flaw.)